engine/resources/exec.go

```diff
@@ -277,6 +285,14 @@ func (obj *ExecRes) CheckApply(apply bool) (bool, error) {
 		}
 		cmd := exec.Command(cmdName, cmdArgs...)
 		cmd.Dir = obj.IfCwd // run program in pwd if ""
+
+		cmdEnv := []string{}
+		for key := range obj.Env {
+			cmdEnv = append(cmdEnv, key+"="+obj.Env[key])
```

Comment:
This actually _does_ do what we want, but it's non-standard and ends up sorting more than we need to. Try and implement the above described solution that *just* sorts the map keys. They're unique (it's a map) so we don't need to sort the values alongside them.

Reply:
Let me make sure I'm understanding correctly. What you're suggesting is
1. Make a list of the keys by iterating through map
2. Sort that list
3. Iterate through sorted list, in each iteration append "<key>=<value>" to cmd.Env

If this is what you mean, isn't it a worse solution to have to use 2 for loops to accomplish the same thing?

Reply:
Yes and no. Technically there's a loop inside the Sort() that you're using, and you're sorting additional stuff which isn't needed.

Reply:
Plus it would be good to get you to learn the standard sort approach when the map keys aren't sortable or are a different type.